packages/core/src/errorCorrection/index.ts

```diff
@@ -0,0 +1,42 @@
+import { Level } from '../level'
+import { ecCodewordsPerVersion } from './ecCodewordsPerVersion'
+import { createGeneratorPolinomial } from './generatorPolinomial'
+import { groupCodewordsInBlocks } from './groupCodewordsInBlocks'
+import { Polinomial } from './Polinomial'
+import { splitIntoCodewords } from './splitIntoCodewords'
+
+export const errorCorrection = (
+  version: number,
+  level: Level,
+  data: string
+) => {
+  const { ecCodewordsPerBlock, group1, group2 } = ecCodewordsPerVersion[
+    `${version}${level}`
+  ]
+
+  const codewords = splitIntoCodewords(data).map(codeword =>
+    parseInt(codeword, 2)
+  )
+
+  const blocks = groupCodewordsInBlocks(codewords, { group1, group2 })
+
+  const generator = createGeneratorPolinomial(ecCodewordsPerBlock)
+
+  const polies = blocks
+    .map(codewords => Polinomial.fromCoef(codewords.slice().reverse()))
+    .map(poly =>
+      // We "prepare" the polinomials so they divide the correct number of times
+      poly.multiply(
+        Polinomial.fromCoefExponents({ [generator.greaterPower()]: 0 })
+      )
+    )
+
+  const ecCodewords = polies
+    .map(poly => poly.modulo(generator))
+    .map(poly => poly.toCoef())
+    // the way the return is expected is from greaterPower -> lesserPower
+    .map(codewords => codewords.reverse())
+
+  // REVIEW: possibly we'll want to return the blocks themselves
+  return ecCodewords
```

Comment:
I'll check what we'll need for the next step, so we can prepare a nice return type here

Reply:
Fixed in 5533dbd